### DISCUSSION TITLE
Get Posts With selected tags using Prisma

### DISCUSSION THREAD

**Discussion by younes-alouani** [prisma/prisma#14859] [Q&A] [answered]
How can I get posts that have selected tags (filter by tag id) ?

Here are the models I'm using :

```
model Post {
  id    Int        @id @default(autoincrement())
  title String
  tags  PostTags[]
}

model PostTags {
  id     Int   @id @default(autoincrement())
  post   Post? @relation(fields: [postId], references: [id])
  tag    Tag?  @relation(fields: [tagId], references: [id])
  postId Int?
  tagId  Int?
}

model Tag {
  id    Int        @id @default(autoincrement())
  name  String     @unique
  posts PostTags[]
}
```
I tried to filter posts using this query but I'm getting also posts that have zero tags :
```
const posts = await prisma.post.findMany({ 
  where: { tags: { every: { id: { in: tags_arrays_ids } } } } 
})
```
I then tried : 
```
const posts = await prisma.post.findMany({
    where: {
      tags: {
        some: {
          id: {
            in: tags_arrays_ids
          }
        }
      }
    }
  })
```
This query will indeed return records having a tag where the id is in the array_ids but i want to get posts where every post has all the tags of the array_ids. Let's assume we have an array of tags [1,2,3] You query will return a post with a single tag id 1. I want to filter posts that have all the tag 1, tag 2 and tag 3 in every post returned

**Comment by austincrim** [comment, marked as answer]
Hey @younes-alouani 👋,

I was able to recreate this as well, and it seems like a bug. Would you mind filling out a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) so our engineers can have a look?

In the meantime, you can use [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw).

**Comment by apecollector** [comment]
Was a bug report ever filled out? I'm having the exact same issue... some other stuff I tried:

```js
const posts = await prisma.posts.findMany({
  where: {
    tags: {
      some: {
        AND: [
          { type: 'Category', value: 'Development' },
          { type: 'Tag', value: 'GitHub' },
        ],
      },
    },
  },
  skip: 0,
  take: 30,
});
```

Returns nothing, while a single item in the AND array does work. 


```js
const posts = await prisma.posts.findMany({
  where: {
    tags: {
      some: {
        AND: [
          { type: 'Category', value: 'Development' },
        ],
      },
    },
  },
  skip: 0,
  take: 30,
});
```

Also OR works fine as expected, however it's not what I want...

```js
const posts = await prisma.posts.findMany({
  where: {
    tags: {
      some: {
        OR: [
          { type: 'Category', value: 'Development' },
          { type: 'Tag', value: 'GitHub' },
        ],
      },
    },
  },
  skip: 0,
  take: 30,
});

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
